packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -2160,24 +2199,147 @@ describe('TokensController', () => {
       });
     });
   });
+
+  describe('when selectedAccountId is not set or account not found', () => {
+    describe('detectTokens', () => {
+      it('updates the token states to empty arrays if the selectedAccountId account is undefined', async () => {
+        await withController(async ({ controller, changeNetwork }) => {
+          ContractMock.mockReturnValue(
+            buildMockEthersERC721Contract({ supportsInterface: false }),
+          );
+          changeNetwork({ selectedNetworkClientId: InfuraNetworkType.sepolia });
+
+          expect(controller.state.tokens).toStrictEqual([]);
+          expect(controller.state.ignoredTokens).toStrictEqual([]);
+          expect(controller.state.detectedTokens).toStrictEqual([]);
+        });
+      });
+    });
+
+    describe('addToken', () => {
+      it('handles undefined selected account', async () => {
+        await withController(async ({ controller, getAccountHandler }) => {
+          getAccountHandler.mockReturnValue(undefined);
+          const contractAddresses = Object.keys(contractMaps);
+          const erc721ContractAddresses = contractAddresses.filter(
+            (contractAddress) => contractMaps[contractAddress].erc721 === true,
+          );
+          const address = erc721ContractAddresses[0];
+          const { symbol, decimals } = contractMaps[address];
+
+          await controller.addToken({ address, symbol, decimals });
+
+          expect(controller.state.tokens).toStrictEqual([
+            {
+              address: '0x9C8fF314C9Bc7F6e59A9d9225Fb22946427eDC03',
+              aggregators: [],
+              decimals: 0,
+              image:
+                'https://static.cx.metamask.io/api/v1/tokenIcons/1/0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03.png',
+              isERC721: true,
+              name: undefined,
+              symbol: 'NOUN',
```

Comment:
I guess those values must match the one defined just above in variables? If yes, we should use the variables here too, to emphasis on this